### ISSUE TITLE
[QUERY] Azure.Blob.Storage auto-renewal of blob

### ISSUE THREAD

**Issue by mmalka** [Azure/azure-sdk-for-net#26048]
nevermind

**Comment by mmalka** [comment]
Here an example of log generated
![image](https://user-images.githubusercontent.com/1161961/147273198-00881826-03bf-4a0a-9f1a-991720a6fc46.png)

**Comment by mmalka** [comment]
@seanmcc-msft
I'm wondering if the changes done on the new generator earlier this year could have introduced a bug where the BlobRestClient would kind of "keep-alive" and keep renewing a lease if it received a partial result on the download start even though it successfully downloaded on the second attempt ?

That would be the explanation to my issue as when I'm in local, I don't have a partial content the first time but the full blob and that's the only difference I can think off right now... on christmas's eve 🤣

I wish you guys an amazing holiday season, hope I can find my answers in 2022.

**Comment by mmalka** [comment]
@seanmcc-msft

I just realized that the DownloadToAsync never *really* complete, it get a Partial Content up to the penultimate byte then, the lease renewal process start running and will spam for about 10 minutes before the azure function shutdown (if no more calls).

Is there a way to force the BlobClient to consider that request closed for good ?

When someone repported about that PartialContent being returned, they were told that it was totally fine, but is it though ? 
https://github.com/Azure/azure-sdk-for-net/issues/17945

Also, someone else started seeing a huge amount of logs about leasing and just ignored it, and was also told it was fine and to just disable logs.
https://github.com/Azure/azure-sdk-for-net/issues/21511

It may be related or not, but I recently updated most of my nuggets (start of december), and arround 12 december, my azure storage subscription started to cost more by a factor of 30, looking at metrics, the insane amount of requests begin there.

If all that is related, that would mean the issue has been introduced somewhere between Azure.Blog.Storage 12.7.0 and 12.10.0.

```
Response [3285bce4-b612-4af4-b38f-a4282c7afeb5] 206 Partial Content (00.2s)
Accept-Ranges:bytes
ETag:"0x8D9XXXXXXXCE454"
Server:Windows-Azure-Blob/1.0,Microsoft-HTTPAPI/2.0
x-ms-request-id:750e3af1-301e-0029-1410-f97f23000000
x-ms-client-request-id:3285bce4-b612-4af4-b38f-a4282c7afeb5
x-ms-version:2020-10-02
x-ms-creation-time:Thu, 26 Nov 2020 13:41:10 GMT
x-ms-blob-content-md5:7eju/BgAi8MTV6QhRPds6A==
x-ms-lease-status:unlocked
x-ms-lease-state:available
x-ms-blob-type:BlockBlob
x-ms-server-encrypted:true
Date:Fri, 24 Dec 2021 21:51:24 GMT
Content-Length:21887503
Content-Type:application/octet-stream
Content-Range:bytes 0-21887502/21887503
Last-Modified:Fri, 24 Dec 2021 21:46:51 GMT
```
